### PR TITLE
Added ep_close for utils provider

### PR DESCRIPTION
**Pull request by alexander-sannikov**
Signed-off-by: Sannikov, Alexander <alexander.sannikov@intel.com>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

File: prov/util/src/util_ep.c [hunk 1]
```diff
@@ -58,3 +58,13 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
 	atomic_inc(&util_domain->ref);
 	return 0;
 }
+
+int ofi_endpoint_close(struct util_ep *util_ep)
+{
+	struct util_domain *util_domain;
+
+	util_domain = util_ep->domain;
+	atomic_dec(&util_domain->ref);
```
Comment (a-ilango) [inline prov/util/src/util_ep.c]: Thanks. There is no need to declare `util_domain` variable. We can decrement the ref directly:
`atomic_dec(&util_ep->domain->ref);`

Comment (alexander-sannikov) [inline prov/util/src/util_ep.c]: Thank you. Fixed.

**Comment by a-ilango** [comment]
Can you please squash the two commits into one?

**Comment by alexander-sannikov** [comment]
Squashed into single commit.

**Pull request merged**